### PR TITLE
feat(rollup-plugin-index-html): support import maps

### PR DESCRIPTION
**Pull request by daKmoR**
- fixes some missing feature in import-maps/resolve

ToDo:
- [x] Read import map from index.html (currently hard coded in test)
- [x] Remove import map from html output?
- [x] Fix failing test in import-maps/resolve
- [x] Cleanup code (console, lint, ...)
- [x] Cleanup history and make nice commits

### CODE REVIEW

File: packages/rollup-plugin-index-html/rollup-plugin-index-html.js [hunk 1]
```diff
@@ -73,6 +76,36 @@ module.exports = (pluginConfig = {}) => {
       return result.rollupOptions;
     },
 
+    resolveId(source, importer) {
+      const { rootDir } = localPluginConfig;
+      const relativeSource = source.replace(rootDir, '');
+
+      const basePath = importer ? importer.replace(rootDir, `${rootDir}::`) : `${rootDir}::`;
+      if (!importMapCache) {
+        // const mapString = fs.readFileSync(path.join(currentDir, 'import-map.json'), 'utf-8');
```
Comment (daKmoR) [inline packages/rollup-plugin-index-html/rollup-plugin-index-html.js]: @LarsDenBakker do you have an idea how to best get the import-map string from the index.html?

also we might need an option to remove the import-map from the generated index.html?

File: packages/building-utils/index-html/extract-resources.js [hunk 2]
```diff
@@ -21,7 +23,11 @@ function extractResources(htmlString) {
 
   const scripts = queryAll(indexHTML, predicates.hasTagName('script'));
   const moduleScripts = scripts.filter(script => getAttribute(script, 'type') === 'module');
+  const importMapScripts = scripts.filter(script => getAttribute(script, 'type') === 'importmap');
+
   const jsModules = [];
+  const importMaps = [];
+  const importMapPaths = [];
```
Comment (LarsDenBakker) [inline packages/building-utils/index-html/extract-resources.js]: I think we need to be a bit clearer around the naming conventions. Maybe `inlineImportMaps` and `importMapPaths` ? And similarly we will need to rename `jsModules` to `inlineModules` and `modulePaths` but we can do that later.

What do you think?

**Comment by LarsDenBakker** [comment]
Should we scope this to the rollup plugin?

File: packages/building-utils/index-html/extract-resources.js (hunk 2 above)
Comment (daKmoR) [reply packages/building-utils/index-html/extract-resources.js]: sounds like a good idea 🤗

**Comment by daKmoR** [comment]
what do you mean by "experimental" flat?
if something changes with the import maps then this package will not require changes as that logic is in `@import-maps/resolve`

**Comment by LarsDenBakker** [comment]
If people use the rollup plugin to set up import maps and either the import map spec or our resolve plugin changes, we break their build. Since import maps are still a work in progress, should we enable this by default or require an explicit opt in for now?

File: packages/building-utils/index-html/extract-resources.js (hunk 2 above)
Comment (daKmoR) [reply packages/building-utils/index-html/extract-resources.js]: changed

**Comment by daKmoR** [comment]
puh you mean like... even if there is an import map in the indx.html we should not respect it unless you start rollup with a specific flag?

imho I doubt there will be big changes... also if you don't have an import map in your index.html it's not used... so I'm not sure if adding a flag makes it much better? it will certainly make it another extra step people need to do/learn... so if it can be avoided I would like that

I mean we could ask in the import-maps repo if they still have something "big" in mind they wanna change?

Overall I think the usage of import-maps will probably not explode overnight 🙈 

Verdict:
I would prefer no flag but if preferred I would be ok with it 🤗

**Comment by LarsDenBakker** [comment]
Yes, yeah let's just do it.

**Pull request merged**